starknet/src/utils/eip712.cairo

```diff
@@ -105,6 +101,7 @@ mod EIP712 {
         ) -> u256 {
             let encoded_data = array![
                 u256 { low: PROPOSE_TYPEHASH_LOW, high: PROPOSE_TYPEHASH_HIGH },
+                Felt252IntoU256::into(starknet::get_tx_info().unbox().chain_id),
```

Comment:
```suggestion
                starknet::get_tx_info().unbox().chain_id.into(),
```

would that work?

Reply:
No because theres a namespace collision between the generic impl and the core lib one

Reply:
love it :sad-pepe: